app.rb

```diff
@@ -0,0 +1,144 @@
+      puts "Title: '#{book.title}', Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'List of all people:'
+    puts(@people.map { |person| "[#{person.type}] Name: #{person.name}, ID: #{person.id} Age: #{person.age}" })
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    choice = gets.chomp.to_i
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+
+    if choice == 1
+      person = create_student(age, name, parent_permission_get, parent_permission_get ? classroom_get : nil)
+    elsif choice == 2
+      person = create_teacher(age, name)
+    else
+      puts 'Invalid choice. Please enter 1 for student or 2 for teacher.'
+      return
+    end
+
+    if person
+      @people << person
+      puts 'Person created successfully'
+    else
+      puts 'Invalid choice. Please enter 1 for student or 2 for teacher.'
+    end
+  end
+
+  def create_student(age, name, parent_permission, classroom)
+    student = Student.new(age, classroom, name: name, parent_permission: parent_permission)
+    student.type = 'Student'
+    student
+  end
+
+  def create_teacher(age, name)
+    puts 'Enter Specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name: name, parent_permission: true)
+    teacher.type = 'Teacher'
+    teacher
+  end
+
+  def parent_permission_get
+    loop do
+      puts 'Has parent permission? [Y/N]:'
+      parent_permission_input = gets.chomp.downcase
+      return parent_permission_input == 'y' if %w[y n].include?(parent_permission_input)
+
+      puts "Invalid input. Please enter 'Y' or 'N' only."
+    end
+  end
+
+  def classroom_get
+    puts 'Enter classroom label:'
+    Classroom.new(gets.chomp)
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+
+    puts 'Author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  def create_rental
+    # List all available books
+    puts 'Available books:'
+    list_all_books
+
+    puts 'Select the index of the book from the available books:'
+    book_index = gets.chomp.to_i
+
+    # List all persons
+    puts 'List of all persons:'
+    list_all_people
+
+    puts "Enter the person's ID from the list of persons:"
+    person_id = gets.chomp.to_i
+
+    puts 'Enter rental date (YYYY-MM-DD):'
+    date = gets.chomp
+
+    # Find the person by their ID
+    person = @people.find { |p| p.id == person_id }
+
+    if person && book_index.between?(0, @books.length - 1)
+      book = @books[book_index]
+
+      rental = Rental.new(date, book, person)
+      @rentals << rental
+      puts " Book ID: #{rental.book.object_id}, Person ID: #{rental.person.object_id}, Rental Date: #{rental.date}."
+    else
+      puts 'Invalid person ID or book selection. Please check the ID and book index and try again.'
+    end
+  end
```

Comment:
Example video
![image](https://github.com/toyman640/school-library/assets/78251927/135e8016-13ed-465d-aa43-c57460d8232f)

Your app (Creating rentals)
![image](https://github.com/toyman640/school-library/assets/78251927/aa7e24c9-dea2-444b-a6cf-5f0cb30fbf89)

- According to the example video, we are required to select from the lists of `Books` and `Persons` using a number (not id)
- Kindly use a numbered list in your app similar to the example video so that it will be easier to select from the list 🛠👍